src/hba.c

```diff
@@ -940,17 +966,50 @@ static bool match_inet6(const struct HBAAddress *haddress, PgAddr *addr)
 	       (src[2] & mask[2]) == base[2] && (src[3] & mask[3]) == base[3];
 }
 
+
 static bool address_match(const struct HBAAddress *haddress, PgAddr *addr)
 {
+	struct addrinfo *gai_result, *gai;
+	int ret;
+	bool found;
+
 	if (haddress->flags & ADDRESS_ALL)
 		return true;
-	switch (haddress->family) {
-	case AF_INET:
-		return match_inet4(haddress, addr);
-	case AF_INET6:
-		return match_inet6(haddress, addr);
-	default:
-		return false;
+	if (!haddress->hostname) {
+		/* match IP address */
+		switch (haddress->family) {
+		case AF_INET:
+			return match_inet4(haddress, addr);
+		case AF_INET6:
+			return match_inet6(haddress, addr);
+		default:
+			return false;
+		}
+	} else {
+		/* match DNS name */
+		ret = getaddrinfo(haddress->hostname, NULL, NULL, &gai_result);
```

Comment:
I'm not sure what problems you ran with our async DNS. But we should definitely try to use that. We have the libraries available. Blocking the event loop is something we should only do in cases where we cannot avoid it.

Reply:
I did attempt to use async DNS but was not able to figure it out. Probably skill issues on my end. Will give it another shot.